File: ANewFolder/MyCards-master (2)/app/src/main/java/edu/univdhaka/cse/cse2216/mycards/java/service/CardService.java
package edu.univdhaka.cse.cse2216.mycards.java.service;

import android.content.Context;

import java.util.List;

import edu.univdhaka.cse.cse2216.mycards.java.domain.Card;
import edu.univdhaka.cse.cse2216.mycards.java.repository.CardRepository;
import edu.univdhaka.cse.cse2216.mycards.java.repository.DatabaseCardRepository;
import edu.univdhaka.cse.cse2216.mycards.java.repository.FirebaseCardRepository;

public class CardService {

    private static final boolean USE_SQLITE_REPOSITORY = false;

    private CardRepository repository;

    public CardService(Context context) {
        repository = USE_SQLITE_REPOSITORY ? new DatabaseCardRepository(context) : new FirebaseCardRepository();
    }

    public List<Card> listCards() {
        return repository.listCards();
    }

    public void listCardsAsync(CardRepository.OnResultListener<List<Card>> resultListener) {
        repository.listCardsAsync(resultListener);
    }

    public void addCard(Card card) {
        repository.addCard(card);
    }

    public void updateCard(Card card) {
        repository.updateCard(card);
    }

    public void deleteCard(Card card) {
        repository.deleteCard(card);
    }
}

File: ANewFolder/MyCards-master (2)/app/src/main/java/edu/univdhaka/cse/cse2216/mycards/kotlin/service/CardService.kt
package edu.univdhaka.cse.cse2216.mycards.kotlin.service

import android.content.Context

import edu.univdhaka.cse.cse2216.mycards.kotlin.domain.Card
import edu.univdhaka.cse.cse2216.mycards.kotlin.repository.CardRepository
import edu.univdhaka.cse.cse2216.mycards.kotlin.repository.DatabaseCardRepository
import edu.univdhaka.cse.cse2216.mycards.kotlin.repository.FirebaseCardRepository

class CardService(context: Context) {

    companion object {

        private val USE_SQLITE_REPOSITORY = true

    }


    private val repository: CardRepository

    init {
        repository = if (USE_SQLITE_REPOSITORY) DatabaseCardRepository(context) else FirebaseCardRepository()
    }

    fun listCards(): List<Card> {
        return repository.listCards()
    }

    fun listCardsAsync(onResult: (List<Card>) -> Unit) {
        repository.listCardsAsync(onResult)
    }

    fun addCard(card: Card) {
        repository.addCard(card)
    }

    fun updateCard(card: Card) {
        repository.updateCard(card)
    }

    fun deleteCard(card: Card) {
        repository.deleteCard(card)
    }
}
